common/element_handle.go

```diff
@@ -1039,35 +1039,35 @@ func (h *ElementHandle) Query(selector string) api.ElementHandle {
 	}
 	result, err := h.evalWithScript(h.ctx, opts, fn, parsedSelector)
 	if err != nil {
-		k6ext.Panic(h.ctx, "querying selector %q: %w", selector, err)
+		return nil, fmt.Errorf("querying selector %q: %w", selector, err)
 	}
 	if result == nil {
-		return nil
+		return nil, fmt.Errorf("querying selector %q", selector)
 	}
-
-	var (
-		handle  = result.(api.JSHandle)
-		element = handle.AsElement()
-	)
-	applySlowMo(h.ctx)
-	if element != nil {
-		return element
+	handle, ok := result.(api.JSHandle)
+	if !ok {
+		return nil, fmt.Errorf("querying selector %q, wrong type %T", selector, result)
 	}
-	handle.Dispose()
-	return nil
+	element := handle.AsElement()
+	if element == nil {
+		handle.Dispose()
+		return nil, fmt.Errorf("querying selector %q", selector)
+	}
+
+	return element, nil
 }
 
 // QueryAll queries element subtree for matching elements.
 // If no element matches the selector, the return value resolves to "null".
-func (h *ElementHandle) QueryAll(selector string) []api.ElementHandle {
+func (h *ElementHandle) QueryAll(selector string) ([]api.ElementHandle, error) {
 	defer applySlowMo(h.ctx)
 
 	handles, err := h.queryAll(selector, h.evalWithScript)
 	if err != nil {
-		k6ext.Panic(h.ctx, "querying all selector %q: %w", selector, err)
+		return nil, fmt.Errorf("querying all selector %q: %w", selector, err)
 	}
 
-	return handles
+	return handles, err
```

Comment:
Should we return `nil` here to be more explicit? As otherwise, if `err` is not `nil` we would have panicked previously.

Reply:
Nice catch, I forgot that one 🤦 Thanks! 🙇

Reply:
@ka3de fixed [here](https://github.com/grafana/xk6-browser/pull/820/commits/f883c1fd0b253c4fcd2ae966d396d14605d87791).